tests/Cms/Fields/FieldMethodsTest.php

```diff
@@ -446,6 +446,53 @@ public function testToUrlEmpty()
 		$this->assertNull($field->toUrl());
 	}
 
+	public function testToUuidUrl()
+	{
+		$app = new App([
+			'roots' => [
+				'index' => $this->tmp
+			],
+			'site' => [
+				'children' => [
+					[
+						'slug' => 'test',
+						'content' => [
+							'title' => 'Test Title',
+							'uuid'  => 'my-page-uuid'
+						],
+						'files' => [
+							[
+								'filename' => 'test.jpg',
+								'content' => [
+									'uuid'  => 'my-file-uuid',
+								]
+							]
+						]
+					]
+				]
+			]
+		]);
+
+		$page = $app->page('page://my-page-uuid');
+		$field = $this->field('page://my-page-uuid');
+		$this->assertSame('/test', $field->toUrl());
+		$this->assertSame($page->url(), $field->toUrl());
+
+		$file = $app->file('file://my-file-uuid');
+		$field = $this->field('file://my-file-uuid');
+		$this->assertSame('/media/pages/test/' . $file->mediaHash() . '/test.jpg', $field->toUrl());
+		$this->assertSame($file->url(), $field->toUrl());
+	}
+
+	public function testToInvalidUuidUrl()
+	{
+		$field = $this->field('page://invalid');
+		$this->assertSame('/', $field->toUrl());
```

Comment:
TBH I feel like this is unexpected. I'd expect `null` here.

Reply:
Hmm, the `url` component always returns a string and it is revealed in this component whether the `uuid` is valid or invalid.

Reply:
You are right, this is a limitation of `Url::to()` and the `url` component.

Thinking about different options, I feel like there is no ideal solution that is both clean and avoids breaking changes.

Returning `null` seems useful, however there would indeed be quite a breaking change in such a central code location.

A few more options I thought about for the "UUID was passed but invalid" case:

3. Throw a `NotFoundException` (more or less backwards-compatible as there was also a PHP error in this case until now)
4. Return the passed UUID string unchanged (keeps the "always a string" return value, but the UUID may not be a useful value in real use)
5. Return `null` from the `url` component but convert that to one of options 1, 3 or 4 in `Url::to()`. Then `Url::to()` could get a new param to control the behavior. But probably needlessly complex.

So far I favor option 3 TBH. It's the least amount of breaking change, but we could still catch the exception in the `toUrl()` field method and return `null` from there if it gets thrown.

What do you think?

Reply:
The difference between `NotFoundException` and returning `null` is that the exception is clearly visible and the `null` value can be missed. So I feel like `NotFoundException` would be safer as it's easier to adapt other parts of the code (both by us and by site and plugin devs).

Reply:
https://github.com/getkirby/kirby/pull/5368/commits/abc8965a7e6cccb54c862ee60f6d8128fbbee8dc

Reply:
Should we declare breaking change?

Reply:
@afbora I don't think we need to declare it a breaking change. UUID support didn't exist in 3.9.x yet.